bigquery_etl/query_scheduling/generate_airflow_dags.py

```diff
@@ -0,0 +1,78 @@
+"""Generates Airflow DAGs for scheduled queries."""
+
+import logging
+import os
+from argparse import ArgumentParser
+from ..util import standard_args
+from bigquery_etl.query_scheduling.dags import Dags
+from bigquery_etl.query_scheduling.task import Task, UnscheduledTask
+
+
+DEFAULT_SQL_DIR = "sql/"
+DEFAULT_DAGS_FILE = "dags.yaml"
+QUERY_FILE = "query.sql"
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument(
+    "--sql_dir",
+    "--sql-dir",
+    help="File or directory containing query and metadata files",
+    default=DEFAULT_SQL_DIR,
+)
+parser.add_argument(
+    "--dags_config",
+    "--dags-config",
+    help="File with DAGs configuration",
+    default=DEFAULT_DAGS_FILE,
+)
+standard_args.add_log_level(parser)
+
+
+def get_dags(sql_dir, dags_config):
+    """Return all configured DAGs including associated tasks."""
+    tasks = []
+
+    # parse metadata.yaml to retrieve scheduling information
+    if os.path.isdir(sql_dir):
+        for root, dirs, files in os.walk(sql_dir):
+            if QUERY_FILE in files:
+                query_file = os.path.join(root, QUERY_FILE)
+
+                try:
+                    task = Task.of_query(query_file)
+                    tasks.append(task)
+                except FileNotFoundError:
+                    # query has no metadata.yaml file; skip
+                    pass
+                except UnscheduledTask:
+                    logging.info(
```

Comment:
If we say "warning" in the message, this should be a `warn`-level message. Also, I would expect that the logging configuration would handle prefixing the output with "WARN".

Reply:
Thinking more about this, I think this should probably be silent about unscheduled tasks, at least for the early phases of this project. I expect it will take a while for us to transition queries to this framework, so initially most queries will not have scheduling metadata.